Add-rfc-0008-used-consensus-algorithm (#49)
internet-sicherheit/eco-blockchain-governance: technical/0008-used-consensus-algorithm/README.md

# 0008: Used consensus algorithm
- Authors: [David Maas](david_maas@hotmail.de) 
- Status: PROPOSED
- Since: 2020-10-21 
- Status Note: Initial Proposal  
- Start Date: 2020-10-21 
- Tags: 

## Summary

> Describes the consensus method used Algorithm.

Reviewer: ```suggestion
Describes the used consensus method/algorithm for the network.
```

## Motivation

> Consensus procedures are a crucial element of any block chain network as they ensure that the integrity and security of these decentralised systems are maintained.

Reviewer: ```suggestion
Consensus procedures are a crucial element of any blockchain network as they ensure that certain integrity and security guarantees of these decentralized systems are maintained.
```

## Specification
The following consensus algorithms are available:

* Proof of Work
* Proof of stake
* Proof of authority (PoA)
* Proof of Importance 

> * Practical Byzantine Fault Tolerance (PBFT) 

Reviewer: This is mixing up different concepts (PoA, PoW, PoS) with concrete algorithms (PBFT).

Author: I have rephrased that a little. There are concepts and implemented algorithms. There are currently a few of these. Should this be more of a recommendation for use or more of a basic description of how the algorithm works?

the algorithm as approved by the TGB MUST be published by the TGB in the official
Sovrin Code Repository and made publicly visible via a web page on the Sovrin
Foundation website.

>[Hyperledger Besu](https://besu.hyperledger.org/en/stable/Concepts/Consensus-Protocols/Overview-Consensus/) implements the following consensus protocols:

>* Ethash (Proof of Work)
>* Clique (Proof of Authority)
>* IBFT 2.0 (Proof of Authority)
>* Quorum IBFT 1.0 (Proof of Authority).

>Comparing Proof of Authority consensus protocols
Besu implements the Clique and IBFT 2.0 Proof of Authority consensus protocols. Proof of Authority consensus protocols work when participants know each other and there is a level of trust between them. For example, in a permissioned consortium network.

>Proof of Authority consensus protocols have faster block times and a much greater transaction throughput than the Ethash Proof of Work consensus protocol used on the Ethereum MainNet.

>In Clique and IBFT 2.0, a group of nodes in the network act as signers (Clique) or validators (IBFT 2.0). The existing nodes in the signer/validator pool vote to add nodes to or remove nodes from the pool.

> 

Reviewer: Those things really supposed to be in quotes?

Author: Yes, because it is also only cited.

## Drawbacks
Depending on the chosen algorithm there are pros and cons. There is no algorithm that achieves the best performance in every respect.

## Prior art
Sovrin provide frameworks and technical documentation that define Node requirements. [Steward Technical and Orgnaizatoinal Policies](https://sovrin.org/wp-content/uploads/Steward-Technical-and-Organizational-Policies-V2.pdf)
Hyperledger Besu implements various consensus algorithms [Overview-Consensus](https://besu.hyperledger.org/en/stable/Concepts/Consensus-Protocols/Overview-Consensus/)

## Unresolved questions
* How can/should the algorithm be optimized?

> * When and how will the algorithm be run?

Reviewer: ```suggestion
```

This is part of the algorithm.

> * How will results of the algorithm be used to adjust network configuration?

Reviewer: IDGI

Author: It is not clear which algorithm is selected for the respective case. Therefore, the question arises as to how the algorithm treats the things. Actually, we know the effects and the load distribution of the algorithms, but this should be taken into account in the selection. So the decision provides the particular solution. You just have to keep that in mind.
